### DISCUSSION TITLE
Knowing if the stream is "live"

### DISCUSSION THREAD

**Discussion by hoschiho** [SRGSSR/pillarbox-android#484] [Q&A]
Hey Pillarbox Team
I'm trying to indicate if a radio stream is live or not. There's the seekToDefaultPosition function, which brings the stream to the live edge, but how do i know if i'm already there? Currently i'm doing a calculation with `player.duration - player.currentPosition`, but this doesn't give me a reliable status. Do you have a better solution for this?

Thanks a lot!
Pascal

**Comment by StaehliJ** [comment]
Hello @hoschiho 

Thanks to have interest of our project.
To know if a content is at live edge if it is your question, you can try:

```kotlin
val window = player.currentTimeline.getWindow(0, Timeline.Window())
val atLiveEdge = player.currentPosition >= window.defaultPositionMs
```
and 

```kotlin
player.isCurrentMediaItemLive
```
to know if the current item is a live stream.

**Reply by hoschiho** [reply]
Thanks for the fast answer @StaehliJ
It seems to work a little bit better, however, after some seconds it still gets out of sync and the boolean jumps. With a threshold it's ok for longer, but once you jump back 10 seconds, you have the same behaviour or it still shows that you're live.

**Comment by StaehliJ** [comment]
Yes for live it can be tricky as the duration and the defaultPosition may change every seconds, you could introduce some threshold to have a more stable experience.

**Reply by StaehliJ** [reply]
@hoschiho I tryed this in the demo to detect live edge. I convert positions in seconds and it seems to be more stable.

```kotlin
val isPositionAtLiveEdge = currentPlaybackPosition.milliseconds.inWholeSeconds >= (currentDefaultPosition - threshold).milliseconds.inWholeSeconds
```
